beacon-chain/p2p/testing/p2p.go

```diff
@@ -47,6 +52,60 @@ func NewTestP2P(t *testing.T) *TestP2P {
 	}
 }
 
+// NewTestP2PWithKey initializes a new p2p test service.
+func NewTestP2PWithKey(t *testing.T, privkey crypto.PrivKey) *TestP2P {
+	ctx := context.Background()
+	h := bhost.NewBlankHost(GenSwarmWithKey(ctx, t, privkey))
+	ps, err := pubsub.NewFloodSub(ctx, h,
+		pubsub.WithMessageSigning(false),
+		pubsub.WithStrictSignatureVerification(false),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return &TestP2P{
+		t:      t,
+		Host:   h,
+		pubsub: ps,
+	}
+}
+
+// GenSwarmWithKey generates a new test swarm. Modified from
+// "github.com/libp2p/go-libp2p-swarm/testing" for
+// internal testing
+func GenSwarmWithKey(ctx context.Context, t *testing.T, privKey crypto.PrivKey) *swarm.Swarm {
```

Comment:
Why do we need this instead of using Identity?

Why is this public?

Reply:
Removed